src/signal.cr

```diff
@@ -72,18 +72,43 @@ require "c/unistd"
   end
 {% end %}
 
-# Signals are processed through the event loop and run in their own Fiber.
-# Signals may be lost if the event loop doesn't run before exit.
-# An uncaught exceptions in a signal handler is a fatal error.
+# This module provides a way to handle OS signals passed to the process.
+#
+# ```
+# puts "Ctrl+C still has the OS default action (stops the program)"
+# sleep 3
+#
+# Signal::INT.trap do
+#   puts "Gotcha!"
+# end
+# puts "Ctrl+C will be caught from now on"
+# sleep 3
+#
+# Signal::INT.reset
+# puts "Ctrl+C is back to the OS default action"
+# sleep 3
+# ```
+#
+# Note:
```

Comment:
That's subjective and it's going' to be properly processed by the doc formatter which requires all caps.